interview-questions.md

```diff
@@ -10,58 +10,70 @@ Then, research the question to expand on your answer. Even if you feel you have
 
   Researched answer:
-
+  React is a library to create components for web applications. It combines javascript and markup language. React is considered flexible, creates a virtual DOM, and a large community to support the technology. We would use React because of its effeciency with updating state of the DOM by having a virtua DOM. The virtual DOM gets updated and sync with the real DOM by ReactDOM. In addition, readibility is easier, since html is used inside javascript. 
 
 
 3. Which lifecycle method is required in a React class component?
 
   Your answer:
+  The required lifecycle method for a class component is Render(). The method Render() is used in the mounting and updating phase of the react lifecyle. Render() allows the visual view of the components in the browser.
 
   Researched answer:
-
+  The render() method is the only required lifecycle method for class components. The method controls the items that will be passed on to the DOM. If items have been passed to the DOM previously, the render() will mutate the DOM with the new updated items.
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
   Your answer:
+  JSX stands for Javascript with XML, one notable difference is JSX uses objects and methods as OOP(object oriented programming), while html is only tags and elements and cannot store information. The use of JSX allows the tracking of properties and any changes, while html main purpose is to render the content in a browser based on its inputs without any storage of information.
 
   Researched answer:
-
+  JSX is known as Javascript Syntax Extension or JavaScript XML. JSX is the combination of Javascript and HTML together. One notable difference between JSX and HTML is that JSX can only return one single element in contrast HTML can return multiple elements.
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
   Your answer:
+  yarn is a package management file system that tracks the dependencies of an application. When yarn is run by itself, we are basically telling the package management file system to download all the dependencies from a yarn log file that is located in the app folder, lists all required dependencies with version. The node module file gets generated afterwards in the app folder and contains all the files for the required dependencies. 
 
   Researched answer:
-
+  Yarn is a package manager for the javascript language. It manages the dependency files of applications. When yarn is run the dependencies that are listed in the package.json are downloaded and a node_modules file is created/updated to store all the dependencies files.
```

Comment:
Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.